src/fetch/payouts.py

```diff
@@ -126,19 +130,31 @@ def from_series(cls, frame: Series) -> RewardAndPenaltyDatum:
             secondary_reward_eth=int(frame["secondary_reward_eth"]),
             secondary_reward_cow=int(frame["secondary_reward_cow"]),
             quote_reward_cow=int(frame["quote_reward_cow"]),
+            service_fee=bool(frame["service_fee"]),
         )
 
     def total_outgoing_eth(self) -> int:
         """Total outgoing amount (including slippage) for the payout."""
-        return self.primary_reward_eth + self.secondary_reward_eth + self.slippage_eth
+        return self.total_eth_reward() + self.slippage_eth
 
     def total_cow_reward(self) -> int:
         """Total outgoing COW token reward"""
-        return self.primary_reward_cow + self.secondary_reward_cow
+        return int(
+            self.reward_scaling()
+            * (self.primary_reward_cow + self.secondary_reward_cow)
+        )
 
     def total_eth_reward(self) -> int:
         """Total outgoing ETH reward"""
-        return self.primary_reward_eth + self.secondary_reward_eth
+        return int(
+            self.reward_scaling()
+            * (self.primary_reward_eth + self.secondary_reward_eth)
+        )
+
+    def reward_scaling(self) -> Fraction:
+        """Scaling factor for service fee
+        The reward is multiplied by this factor"""
+        return 1 - SERVICE_FEE_FACTOR * self.service_fee
```

Comment:
Interesting that this works....

Reply:
(sorry about that, the more natural version produced type errors for me)